gensim/models/nmf.py

```diff
@@ -177,7 +264,7 @@ def show_topics(self, num_topics=10, num_words=10, log=False,
         if normalize is None:
             normalize = self.normalize
 
-        sparsity = self._W.getnnz(axis=0)
+        sparsity = (self._W == 0).mean(axis=0)
```

Comment:
Mean of zeros? What does this do?

Reply:
OK, got it. So `sparsity` is an array of densities of the columns.

BTW this `sparse_matrix == 0` prints a warning in my scipy 0.19.0:

```
/Volumes/work/workspace/vew/pii3.6/lib/python3.6/site-packages/scipy/sparse/compressed.py:214: SparseEfficiencyWarning: Comparing a sparse matrix with 0 using == is inefficient, try using != instead.
  ", try using != instead.", SparseEfficiencyWarning)
```

Reply:
I just realized that `self._W` is actually dense, thus there will be no warning.

Reply:
Careful about temporary copies and RAM then (each sub-expression like `1 - (D + 1))` can create an independent array copy).

Reply:
Yes, that seems more efficient.

But remember to include a clear comment, otherwise we'll be scratching our heads in a few months, wondering what this construct is for.